### DISCUSSION TITLE
Browser memory shoot up

### DISCUSSION THREAD

**Discussion by kishorekaruppusamy** [pyodide/pyodide#4177] [Q&A] [answered]
I'm trying to use Pyodide to build a random forest model, so I loaded the pandas and scikit-learn packages.
My browser's memory increases to 250–300 MB while I'm trying to load these packages.

Can i reduce this in some way ??

<img width="985" alt="Screenshot 2023-09-26 at 3 38 35 PM" src="https://github.com/pyodide/pyodide/assets/103121724/cb30ecb7-1a66-42ae-a770-ec3e340a3810">

**Comment by rth** [comment, marked as answer]
What's the memory usage outside of the browser for this model? Depending on the data / model size I wouldn't be surprised by those numbers.

Also we use the in-memory MEMFS file system, which then counts as memory used by the browser. For scikit-learn + dependencies, I would estimate the size of files between 50-100 MB. 

If memory is really an issue, you could try to export your model as ONNX and use their WASM runtime which should be more lightweight, while still doing the pre-/post-processing in Python.

**Comment by kishorekaruppusamy** [comment]
If i am not wrong this 250 Mb is the size of python packages while installing Right ??
